src/averageAges.js

```diff
@@ -34,7 +43,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenAge = people
+    .filter(
+      woman => withChildren
+        ? woman.sex === 'f' && people.some(human => human.mother === woman.name)
```

Comment:
no need to check woman.sex here (line 49), if she's a mother she' s definitely a woman